File: Week 1/sort_array_0_1_2.cpp
//Given an array of size N containing 0s, 1s, and 2s; sort the array in ascending order.

#include<bits/stdc++.h>
using namespace std;
void show_array(int arr[],int size)
{
	for(int i=0;i<size;i++)
	{
		cout<<arr[i]<<" ";
	}
}

void sort_array(int arr[],int size)
{
	int count0=0, count1=0,count2=0;
	for(int i=0;i<size;i++)
	{
		switch(arr[i])
		{
			case 0:
				count0++;
				break;
			case 1:
				count1++;
				break;
			case 2:
				count2++;
				break;
		}
	}
	int i=0;
	while(count0>0)
	{
		arr[i++] = 0;
		count0--;
	}
	while(count1>0)
	{
		arr[i++] = 1;
		count1--;
	}
	while(count2>0)
	{
		arr[i++] = 2;
		count2--;
	}
	show_array(arr,size);
}

int main()

{
	int n,arr[n];
	cout<<"Enter Size of an Array:";
	cin>>n;
	cout<<"Enter the Elements(only o,1 and 2) in an Array"<<endl;
	for(int i=0;i<n;i++)
	{
		cin>>arr[i];
	}
	cout<<"The Array is ";
	show_array(arr,n);
	cout<<endl;
	cout<<"The Sorted Array is ";
	sort_array(arr,n);
}

File: Week 1/sort_array_0_1_2.py
#Given an array of size N containing 0s, 1s, and 2s; sort the array in ascending order.
arr = [0,1,0,0,1,1,2,1,2]
print("The Array is ", arr)
arr_sort = []
cnt0,cnt1,cnt2 = 0,0,0
for i in arr:
    if i == 0:
        cnt0 +=1
    elif i == 1:
        cnt1 +=1
    else:
        cnt2 +=1

while(cnt0>0):
    arr_sort.append(0)
    cnt0 -=1
while(cnt1>0):
    arr_sort.append(1)
    cnt1 -=1
while(cnt2>0):
    arr_sort.append(2)
    cnt2 -=1

print("The Sorted Array is ", arr_sort)